pmm-app/src/pmm-qan/panel/QueryAnalytics.constants.test.tsx

```diff
@@ -0,0 +1,552 @@
+import { METRIC_CATALOGUE } from './QueryAnalytics.constants';
+
+const metrics = {
+  num_queries: {
+    sum: 100,
+    avg: 10,
+  },
+  query_time: {
+    sum: 100,
+    avg: 10,
+  },
+  bytes_sent: {
+    sum: 200,
+  },
+  rows_sent: {
+    sum: 100,
+  },
+  rows_examined: {
+    sum: 100,
+  },
+  docs_scanned: {
+    sum: 200,
+  },
+  docs_returned: {
+    sum: 100,
+  },
+  filesort: {
+    sum: 200,
+  },
+  filesort_on_disk: {
+    sum: 200,
+  },
+  full_join: {
+    sum: 100,
+  },
+  full_scan: {
+    sum: 100,
+  },
+  tmp_disk_tables: {
+    sum: 100,
+  },
+  tmp_table_on_disk: {
+    sum: 100,
+  },
+  tmp_table: {
+    sum: 100,
+  },
+  tmp_tables: {
+    sum: 100,
+  },
+  tmp_table_sizes: {
+    sum: 100,
+  },
+  innodb_io_r_bytes: {
+    sum: 100,
+  },
+  innodb_io_r_ops: {
+    sum: 100,
+  },
+  innodb_io_r_wait: {
+    sum: 100,
+    avg: 10,
+  },
+  innodb_queue_wait: {
+    sum: 100,
+    avg: 10,
+  },
+  innodb_rec_lock_wait: {
+    sum: 100,
+    avg: 10,
+  },
+  lock_time: {
+    sum: 100,
+    avg: 10,
+  },
+  merge_passes: {
+    sum: 100,
+    avg: 10,
+  },
+  qc_hit: {
+    sum: 100,
+    avg: 10,
+  },
+  response_length: {
+    sum: 100,
+    avg: 10,
+  },
+};
+
+describe('Query analytics metrics::', () => {
+  it('bytes_sent', () => {
+    const metric = METRIC_CATALOGUE.bytes_sent;
+
+    const processedMetric = metric.metricRelation(metrics);
+    const absentMetric = metric.metricRelation({});
+
+    expect(absentMetric).toBe('');
+    expect(processedMetric).toBe('2.00 Bytes per row sent');
+  });
+
+  it('blk_read_time', () => {
+    // Metric with empty relation
```

Comment:
I see a lot of duplicate tests with the comment `// Metric with empty relation`, maybe would've been better to create an array with such metrics and add a single test with a loop to save you time and make it easier to maintain.

Reply:
makes sense 👍  replaced with test.each